apps/directory/src/views/NetworkReport.vue

```diff
@@ -0,0 +1,167 @@
+                        </div>
+                      </div>
+                    </div>
+                  </Tab>
+                  <Tab :title="`Biobanks (${biobanks.length})`">
+                    <div class="pt-3">
+                      <div
+                        v-for="(biobank, index) in biobanks"
+                        :key="biobank.id"
+                      >
+                        <hr v-if="index" />
+                        <h4>
+                          <router-link :to="`/biobank/${biobank.id}`">
+                            {{ biobank.name }}
+                          </router-link>
+                        </h4>
+                        <ReportDescription
+                          :description="biobank.description"
+                          :maxLength="250"
+                        />
+                      </div>
+                    </div>
+                  </Tab>
+                </Tabs>
+              </div>
+              <!-- Right side card -->
+              <div class="col-md-4">
+                <div class="card">
+                  <div class="card-body">
+                    <div class="card-text">
+                      <h5>Contact Information</h5>
+                      <ReportDetailsList :reportDetails="contact" />
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { Spinner, Tab, Tabs } from "molgenis-components";
+import { computed, ref } from "vue";
+import { useRoute } from "vue-router";
+import ViewGenerator from "../components/generators/ViewGenerator.vue";
+import Breadcrumb from "../components/micro-components/BreadcrumbComponent.vue";
+import CollectionTitle from "../components/report-components/CollectionTitle.vue";
+import ReportDescription from "../components/report-components/ReportDescription.vue";
+import ReportDetailsList from "../components/report-components/ReportDetailsList.vue";
+import ReportTitle from "../components/report-components/ReportTitle.vue";
+import {
+  getCollectionDetails,
+  mapContactInfo,
+} from "../functions/viewmodelMapper";
+import { useNetworkStore } from "../stores/networkStore";
+import { useQualitiesStore } from "../stores/qualitiesStore";
+import { useSettingsStore } from "../stores/settingsStore";
+
+const route = useRoute();
+
+let loaded = ref(false);
+
+const settingsStore = useSettingsStore();
+const networkStore = useNetworkStore();
+const qualitiesStore = useQualitiesStore();
+
+const networkPromise = networkStore.loadNetworkReport(route.params.id);
+const qualitiesPromise = qualitiesStore.getQualityStandardInformation();
+Promise.all([qualitiesPromise, networkPromise]).then(
+  () => (loaded.value = true)
+);
+
+const uiText = computed(() => settingsStore.uiText);
+const networkReport = computed(() => networkStore.networkReport);
```

Comment:
is this really how you like it? I don't have problems but in other catalogue we have abstracted this away in typed display components. I think that is nicer.

Reply:
what do you mean with typed display components?

Reply:
Also not sure what you mean. But this is mostly a copy-pasta from the old code with some editing to make it work, so not much thought went into restructuring how the components tie together.